standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java

```diff
@@ -870,44 +870,88 @@ private boolean isViewTable(String catName, String dbName, String tblName) throw
       List<? extends Object> paramsForFilter, List<String> joinsForFilter, Integer max)
       throws MetaException {
     boolean doTrace = LOG.isDebugEnabled();
+
+    // The JDBC client driver implementations of Derby, PostgreSQL, MySQL, MariaDB and Oracle send the number of
+    // parameters being set in a executed statement as a 2 byte signed integer to the servers. This indirectly
+    // sets s limit on the number of parameters that can be set to 32767. This is also the number of parameters
+    // that can be passed to the JDO executeArray call.
+    final int JDBC_STMT_PARAM_LIMIT = 32767;
```

Comment:
2 thoughts on this
1) Can we move this outside this method to the top of the class? 
2) I understand that based on the research all DBs support atleast 32767 parameters. But do we really know that it works? We dont have a test case for this that tests across all databases nor have we tested this, afaik. Also we are starting to support a pluggable model for DB support. So users can use HMS with other DBs as well for example DB2 (all they have to do is provide the schema scripts). So we don't know what their limits would be.
Would it make sense to use a smaller number? what is the downside of doing so?

Reply:
1. I am OK with this. Will do this and resubmit the patch.

> 2. But do we really know that it works? We dont have a test case for this that tests across all databases nor have we 
> tested this, afaik.

Agreed. I will look into creating a test. Insofar as the issue itself is concerned they seem to be documented limitations of the JDBC driver and the solution is to break the original query into multiple queries. Please do take a look at - https://www.postgresql.org/message-id/16832734.post%40talk.nabble.com

> We dont have a test case for this that tests across all databases nor have we tested this, afaik.

Agreed. I can work on this.

> Also we are starting to support a pluggable model for DB support. So users can use HMS with other DBs as well for 
> example DB2 (all they have to do is provide the schema scripts). So we don't know what their limits would be.
> Would it make sense to use a smaller number? what is the downside of doing so?

Thank you for explaining about the pluggable model for DB support. We can consider a smaller size for the parameters that are passed to the query.

The downside of reducing the number of parameters that are passed would be that it can potentially result in increases the number of iterations required to get the result of the query for the same number of parameters.

for example if there are 30000 parameters, with a size of 32767, it can be executed in a single query. But if we set the size to 1000, it will result in 30000/1000 = 30 iterations.

This will result in increased latency.

It is important to note that this part of the code is executed for all DDL, DML and queries that involved partitioned tables. Hence it has potential to increase the latency across the spectrum.